### ISSUE TITLE
[SRanipal] Resonite doesn't re-interlise eye tracking

### ISSUE THREAD

**Issue by darbdarb** [Yellow-Dog-Man/Resonite-Issues#3390]
### Describe the bug?

When eye tracking disconnects resonite doesn't re-initialise eye tracking, sometimes even face tracking but face tracking will come back

### To Reproduce

1. Launch Resonite with a VR Headset that uses SRanipal i.e Vive Pro Eye
2. Resonite will launch SRanipal and starts to initialise eye and/or face tracking
3. Once in world disconnect headset via link box button or cable/ or close SRanipal on taskbar and resonite will relaunch it
4. Reconnect Headset / once SRanipal relaunches you'll notice below
5. Sranipal will show eyes as orange. Sometimes face tracking will stay Ref Image is no resonite loaded ![Image](https://github.com/user-attachments/assets/9c6e5866-0dd6-41e9-99c2-d0f45aeca640)

### Expected behavior

I excpect SRanipal to relaunch eye and/or face tracking like it does on startup without having to reboot resonite 

### Resonite Version Number

2024.12.17.538

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Vive Pro Eye with vive facial tracker

### Log Files

[DARBDARB - 2024.12.17.538 - 2024-12-18 11_44_42.log](https://github.com/user-attachments/files/18173056/DARBDARB.-.2024.12.17.538.-.2024-12-18.11_44_42.log)

**Comment by shiftyscales** [comment]
> I excpect SRanipal to relaunch eye and/or face tracking like it does on startup without having to reboot resonite

Based on the order of operations you provided above- it sounds like the headset is still disconnected after Resonite has restarted SRanipal? If you have the headset already on- does Resonite properly re-initialize SRanipal?

Resonite should already attempt to reconnect to the facial / eye tracker if the headset is ever disconnected. Is there a reason you are forcibly closing SRanipal while Resonite is still running / after the headset has been disconnected / before the headset is reconnected fully?

**Comment by kulzae** [comment]
Adding that Resonite gets “stuck” with whatever data was last sent by the eye tracker rather than acting as if the tracker is disconnected. Often disconnecting the tracking is unintentional and happens often if using the pro eye with the wireless module as having to swap the battery will disconnect the headset and cause this issue

**Comment by darbdarb** [comment]
> > I expect SRanipal to relaunch eye and/or face tracking like it does on startup without having to reboot resonite
> 
> Based on the order of operations you provided above- it sounds like the headset is still disconnected after Resonite has restarted SRanipal? If you have the headset already on- does Resonite properly re-initialize SRanipal?
> 
> Resonite should already attempt to reconnect to the facial / eye tracker if the headset is ever disconnected. Is there a reason you are forcibly closing SRanipal while Resonite is still running / after the headset has been disconnected / before the headset is reconnected fully?

Say if ur headset has a faulty connection, but u can fix it by unplugging and unplugging ur headset, u have to relaunch sranipal to reinstate the tracking module eye chip, now u have no eye tracking

**Comment by shiftyscales** [comment]
I understand why it can happen- but I what I was more meaning- when it does happen- is the result the same if you restart SRanipal while the headset is running  / connected fully as opposed to restarting SRanipal **then** powering on / connecting the headset, @kulzae @darbdarb?

**Comment by darbdarb** [comment]
> I understand why it can happen- but I what I was more meaning- when it does happen- is the result the same if you restart SRanipal while the headset is running  / connected fully as opposed to restarting SRanipal **then** powering on / connecting the headset, @kulzae @darbdarb?

Yes if you restart SRanipal while the headset is connected u get the same result.

**Comment by kulzae** [comment]
> I understand why it can happen- but I what I was more meaning- when it does happen- is the result the same if you restart SRanipal while the headset is running / connected fully as opposed to restarting SRanipal **then** powering on / connecting the headset, [@kulzae](https://github.com/kulzae) [@darbdarb](https://github.com/darbdarb)?

I just tested and restarting sranipal while the headset is disconnected does not change the result. the eyes remain in the state they were in when the headset was disconnected in resonite and the sranipal icon stays with the eyes in the "orange" state

[RENEGADE - 2025.2.21.98 - 2025-02-24 19_47_15.log](https://github.com/user-attachments/files/18952023/RENEGADE.-.2025.2.21.98.-.2025-02-24.19_47_15.log)

**Comment by kulzae** [comment]
Tested again this time restarting sranipal after the headset has been reconnected and it seems to reinitialize the eye tracking (and freezes resonite for a while when it does) so its likely something to do with sranipal rather than something resonite is doing. Resonite attempts to automatically restart sranipal every time is is closed
